radicle-dag/src/lib.rs

```diff
@@ -130,6 +130,21 @@ impl<K: Eq + Copy + Hash, V> Dag<K, V> {
             .filter_map(|k| self.graph.get(k).map(|n| (k, n)))
     }
 
+    /// Merge a DAG into this one.
+    ///
+    /// If a key exists in both graphs, its value is set to that of the other graph.
+    pub fn merge(&mut self, other: Self) {
+        for k in other.tips.into_iter() {
+            self.tips.insert(k);
+        }
+        for k in other.roots.into_iter() {
+            self.roots.insert(k);
+        }
+        for (k, v) in other.graph.into_iter() {
+            self.graph.insert(k, v);
+        }
+    }
```

Comment:
This makes me wonder if there's a better use for a `merge_with` that uses a comparison function on the keys, essentially allowing the selection of a later key in a graph if we're involving clocks and such.

Reply:
Hmm yeah I was thinking *should Dag be a CRDT?*

If it were, we'd basically constrain keys to `Semilattice`, and then all inserts would merge. But I'm not sure if there's a use-case, because the way we use the DAG is that keys are `Oid` and therefore should only be overwritten if they point to the same commit. So in practice values should never be changed after a merge, only added.

Reply:
Makes sense! Thanks for the explanation :)